### ISSUE TITLE
Broken Playtime

### ISSUE THREAD

**Issue by xFatix** [plan-player-analytics/Plan#1424]
Describe the issue
Yesterday i wanted to see my playtime on the server (spigot). Then I saw that i have 6 months of total playtime but this isnt possible because the server is ~2 weeks old. This happend to nearly every user on my server. 

Here is the database.db i cant upload it here.
https://anonfile.com/zdp3Scv1ob/database_db

Additional information
![image](https://user-images.githubusercontent.com/23125628/80957521-98556200-8e03-11ea-848b-83dbc3387d07.png)

**Comment by Rsl1122** [comment]
Possibly related to #1396

**Comment by Rsl1122** [comment]
The database contains multiple sessions (42) that are over four days long, but new sessions were started some minutes apart..

Here is a screenshot from a player page of a single user
![kuva](https://user-images.githubusercontent.com/24460436/80983336-20079480-8e35-11ea-9a0c-f9483b92e007.png)

All the affected sessions end on 3.5. Sunday 20:23 (GMT+3, my pc timezone) - I think the server shut down at this time.
![kuva](https://user-images.githubusercontent.com/24460436/80985079-68c04d00-8e37-11ea-9a52-694bf760f630.png)
![kuva](https://user-images.githubusercontent.com/24460436/80985165-8392c180-8e37-11ea-96f1-706515db75f1.png)
a lot of them end within the same second
![kuva](https://user-images.githubusercontent.com/24460436/80985352-c6549980-8e37-11ea-8751-2c2376a120ad.png)
The TPS data confirms that server was down for about 3 minutes around this time
![kuva](https://user-images.githubusercontent.com/24460436/80985735-4549d200-8e38-11ea-8484-955262147f4b.png)
(~280 000ms, 4.7 minutes)

----

I'm still trying to figure out what excactly happened, but here's what I'm currently investigating..

- The possibility that something was blocking PlayerJoinEvent code before cacheSession call (Ends sessions in progress if one is already in progress)
- The possibility that something was blocking PlayerQuitEvent code before endSession call
- The possibility that some sessions were not removed from the Active session cache on removal and server shutdown save saved these sessions

----

`plan_world_times` reflects the same oddity as sessions table - the world times don't differ from the session length
```sql
SELECT s.id, ((session_end - session_start) - (SUM(survival_time+creative_time+adventure_time+spectator_time))) as diff, session_end
FROM plan_sessions s
INNER JOIN plan_world_times w on s.id=w.session_id
GROUP BY s.id;
```
![kuva](https://user-images.githubusercontent.com/24460436/80990121-90ff7a00-8e3e-11ea-9fcb-a34845d77a71.png)

This isn't surprising as they are updated at the same time as sessions are ended.

----

**Please let me know if any more of these show up on next reboots.**

**Comment by Rsl1122** [comment]
Issue occurred again in build 904 (reported on discord)

![image](https://user-images.githubusercontent.com/24460436/107126480-e75f8480-68b8-11eb-8451-d8daafb0d276.png)

```
Plan build 904, using GeyserMC

31.01:	
-- server restart at 10:31
[10:31:46] [Server thread/INFO]: [Plan] Enabling Plan v5.2 build 904
-- 17:12 Plan reloaded [17:12:49] [Server thread/INFO]: [Plan] Enabling Plan v5.2 build 904
-- 17:14 Plan reloaded [17:14:47] [Server thread/INFO]: [Plan] Enabling Plan v5.2 build 904
-- 17:15 Plan reloaded [17:15:40] [Server thread/INFO]: [Plan] Enabling Plan v5.2 build 904
-- 17:16 Plan reloaded [17:16:43] [Server thread/INFO]: [Plan] Enabling Plan v5.2 build 904
-- 17:17 Plan reloaded [17:17:46] [Server thread/INFO]: [Plan] Enabling Plan v5.2 build 904
-- 17:18 Plan reloaded [17:18:23] [Server thread/INFO]: [Plan] Enabling Plan v5.2 build 904

[17:30:39] [User Authenticator #47/INFO]: UUID of player [PLAYER] is [UUID]
-- 18:02 Plan reloaded [18:02:05] [Server thread/INFO]: [Plan] Enabling Plan v5.2 build 904
[18:20:59] [Server thread/INFO]: [PLAYER] left the game
-- Server log terminates at 22:31

01.02:
[04:05:17] [User Authenticator #60/INFO]: UUID of player [PLAYER] is [UUID]
[04:34:59] [Server thread/INFO]: [PLAYER] left the game
[04:41:14] [User Authenticator #61/INFO]: UUID of player [PLAYER] is [UUID]
[05:14:24] [Server thread/INFO]: [PLAYER] left the game
-- End of session is set as 01.02.2021 05:46, length of 12:15:35, calculated start 31.01 17:30:39 ! Note: SESSION IS NOT REALLY LIVE !
[05:46:14] [Server thread/INFO]: Stopping the server
[05:46:14] [Server thread/INFO]: [Plan] Disabling Plan v5.2 build 904
[05:46:14] [Server thread/INFO]: [Plan] Processing critical unprocessed tasks. (0)
[05:46:14] [Server thread/INFO]: [Plan] Processing complete.
[05:46:14] [Server thread/INFO]: [Plan] Webserver has been disabled.
[05:46:14] [Server thread/INFO]: [Plan] Player Analytics Disabled.
-- Server shutdown at [05:46:14]
```

Hypothesis confirmed: The broken session was stored during shutdown.

What went wrong: unknown, it is possible that the reloads somehow affected the issue.

- Could the session cache somehow have kept the sessions in memory due to reloads?
- Could the reload during a session have affected the sessions?
- Why did the consequent leaves and joins not store the session in the database?
- Were other sessions also affected?

**Comment by AuroraLS3** [comment]
Possibly fixed in 5.2 build 1115. There was some code that had odd variable names. I don't really know if this has been fixed, but the code around session storage has changed in some ways that might prevent this.
